server/controllers/bookController.js

```diff
@@ -61,4 +62,61 @@ const deleteSingleBook = async (req, res) => {
     }
 };
 
-export default { storeBooks, getAllBooks, getSingleBook, deleteSingleBook };
+const requestBook = async (req, res) => {
+    const { id, email } = req.user;
+
+    await LendingRequest.query().insert({
+        user: id,
+        book: req.params.id,
+        status: 'pending',
+        requestDate: new Date()
+    });
+
+    return res.status(200).jsend({
+        message: 'Request received, you would be notified when approved'
+    });
+};
+
+const decideBookRequest = async (req, res) => {
+    const { id, email } = req.user;
+
+    await LendingRequest.query()
+        .patch({
+            status: 'approved',
+            approvedDate: new Date(),
+            returned: false,
+            returnDate: new Date(new Date().setDate(new Date().getDate() + 30))
```

Comment:
Just for future use, when next working with dates (most especially on the backend), reach out for a library like `moment.js`. On the frontend it might be heavy but on the backend we do not have that concern

Reply:
Alright, I would take that into practice. I would change it during the next fix to this branch after Kelvin.